clients/filesystem-fuse/src/fuse_api_handle.rs

```diff
@@ -0,0 +1,448 @@
+        size: u32,
+    ) -> fuse3::Result<ReplyData> {
+        let data = self.local_fs.read(inode, fh, offset, size).await?;
+        Ok(ReplyData { data: data })
+    }
+
+    async fn write(
+        &self,
+        _req: Request,
+        inode: Inode,
+        fh: u64,
+        offset: u64,
+        data: &[u8],
+        _write_flags: u32,
+        _flags: u32,
+    ) -> fuse3::Result<ReplyWrite> {
+        let written = self.local_fs.write(inode, fh, offset, data).await?;
+        Ok(ReplyWrite { written: written })
+    }
+
+    async fn statfs(&self, _req: Request, _inode: Inode) -> fuse3::Result<ReplyStatFs> {
+        //TODO: Implement statfs for the filesystem
+        Ok(ReplyStatFs {
+            blocks: 1000000,
+            bfree: 1000000,
+            bavail: 1000000,
+            files: 1000000,
+            ffree: 1000000,
+            bsize: 4096,
+            namelen: 255,
+            frsize: 4096,
+        })
+    }
+
+    async fn release(
+        &self,
+        _eq: Request,
+        inode: Inode,
+        fh: u64,
+        _flags: u32,
+        _lock_owner: u64,
+        _flush: bool,
+    ) -> fuse3::Result<()> {
+        self.local_fs.close_file(inode, fh).await
+    }
+
+    async fn opendir(&self, _req: Request, inode: Inode, flags: u32) -> fuse3::Result<ReplyOpen> {
+        let file_handle = self.local_fs.open_dir(inode, flags).await?;
+        Ok(ReplyOpen {
+            fh: file_handle.handle_id,
+            flags: flags,
+        })
+    }
+
+    type DirEntryStream<'a>
+        = BoxStream<'a, fuse3::Result<DirectoryEntry>>
+    where
+        T: 'a;
+
+    #[allow(clippy::needless_lifetimes)]
+    async fn readdir<'a>(
+        &'a self,
+        _req: Request,
+        parent: Inode,
+        _fh: u64,
+        offset: i64,
+    ) -> fuse3::Result<ReplyDirectory<Self::DirEntryStream<'a>>> {
+        let current = self.local_fs.stat(parent).await?;
+        let files = self.local_fs.read_dir(parent).await?;
+        let entries_stream =
+            stream::iter(files.into_iter().enumerate().map(|(index, file_stat)| {
+                Ok(DirectoryEntry {
+                    inode: file_stat.file_id,
+                    name: file_stat.name.clone().into(),
+                    kind: file_stat.kind,
+                    offset: (index + 3) as i64,
+                })
+            }));
+
+        let relative_paths = stream::iter([
+            Ok(DirectoryEntry {
+                inode: current.file_id,
+                name: ".".into(),
+                kind: Directory,
+                offset: 1,
+            }),
+            Ok(DirectoryEntry {
+                inode: current.parent_file_id,
+                name: "..".into(),
+                kind: Directory,
+                offset: 2,
+            }),
+        ]);
+
+        let combined_stream = relative_paths.chain(entries_stream);
+        Ok(ReplyDirectory {
+            entries: combined_stream.skip(offset as usize).boxed(),
```

Comment:
The current implement is a little hack, any better solution?

Reply:
We can improve in the other pr.